test/e2e/azure_test.go

```diff
@@ -323,6 +323,11 @@ var _ = Describe("Workload cluster creation", func() {
 				WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
 				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 				WaitForMachinePools:          e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+				// nvidia-gpu flavor creates a config map as part of a crs, that exceeds the annotations size limit when we do kubectl apply.
```

Comment:
how come we're not running into this when using the flavor outside of e2e? Is there a chance users will run into this if using `clusterctl` with --flavor gpu?

Reply:
That's a great point. I think it should be updated in the docs. Will do it now.